io/src/lib.rs

```diff
@@ -65,6 +80,27 @@ pub struct Take<'a, R: Read + ?Sized> {
     remaining: u64,
 }
 
+impl<'a, R: Read + ?Sized> Take<'a, R> {
+    #[cfg(any(feature = "alloc", feature = "std"))]
```

Comment:
Side note: the way we do `std` and `alloc` features is totally convoluted. Looks like this was copied from `serde` which had to do it to maintain compatibility because it originally didn't have the `alloc` feature because of MSRV. This is not the case for our library, so we could've safely set `std = ["alloc"]` and unify everything.

But changing it now would be a big refactoring for `io` crate which I hope will go away anyway. So let's not do it unless for some unforeseen reason we find out that keeping the crate is better in which case we can clean this up.

Reply:
I hope once `push_decode` is in there will be no reason for any IO other than thin helpers for lazy people (where we'd use `std`/`tokio`/`async-std`) and `no_std` people can either pull in `lgio` or write their own bridge (it's pretty simple anyway, < 15 LOC for sync version). And in case I'm missing something then `lgio` provides statically analyzable errors at the expense of having to insert an additional method in some rare circumstances which I think is a better tradeoff.

Reply:
Oo, this all sounds great. OK, I hope you are right that this is possible and doable on the timeline we want.